src/store/Network.ts

```diff
@@ -251,20 +253,56 @@ export default {
             await Lock.uninitialize(callback, { getters });
         },
 
-        async CONNECT({ commit, dispatch, getters, rootGetters }, newCandidate: string | undefined) {
+        async CONNECT({ dispatch, rootGetters }, newCandidate: string | undefined) {
             const currentProfile: ProfileModel = rootGetters['profile/currentProfile'];
             const networkService = new NetworkService();
-            const nodeService = new NodeService();
-            const networkModelResult = await networkService
-                .getNetworkModel(newCandidate, (currentProfile && currentProfile.generationHash) || undefined)
-                .toPromise();
-            if (!networkModelResult) {
-                throw new Error('Connect error, active peer cannot be found');
-            }
-            const { networkModel, repositoryFactory, fallback } = networkModelResult;
-            if (fallback) {
-                throw new Error('Connection Error.');
+            let nodeNetworkModelResult: any;
+            if (newCandidate) {
+                nodeNetworkModelResult = await networkService
+                    .getNetworkModel(newCandidate, (currentProfile && currentProfile.generationHash) || undefined)
```

Comment:
If we don't change the current implementation of `networkService.getNetworkModel` it'll find some other node which is available but we don't want this in this block. This block is for 'I have a custom node let's try that and if doesn't work give me an error' .